presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/CatalogSchemaAccessControlRule.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.base.security;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonUnwrapped;
+import io.prestosql.spi.connector.CatalogSchemaName;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class CatalogSchemaAccessControlRule
+{
+    private final Optional<Pattern> catalogRegex;
+    private SchemaAccessControlRule schemaAccessControlRule;
```

Comment:
I would remove it from here and possibly rename it to `CatalogAccessRulePattern`. Then basing on `CatalogAccessRulePattern` and `SchemaAccessRulePattern` I would build `CatalogSchemaAccessControlRule`. No setters and partially build objects.

Reply:
I changed the constructor to just explicitly have arguments of the nested type.  Hopefully we can simplify this later, but it works for now.